src/Dock.Model/FactoryBase.cs

```diff
@@ -316,19 +334,27 @@ public void SetFocusedDockable(IDock dock, IDockable dockable)
             return null;
         }
 
-        private IPinDock PinClosestPinDock(IRootDock root, IDockable dockable)
+        private IPinDock? FindClosestPinDock(IDockable dockable)
         {
-            // TODO: Find closest pin dock to the dockable (Top, Bottom, Left or Right).
-
-            if (root.Left is null)
+            while (dockable.Owner != null)
             {
-                root.Left = CreatePinDock();
-                root.Left.Alignment = Alignment.Left;
-                root.Left.IsExpanded = false;
-                root.Left.AutoHide = true;
+                if (dockable.Owner is IRootDock dock)
+                {
+                    // TODO: Choose most revelant dock, currently random for demonstration purposes
+                    var docks = new[] { dock.Left, dock.Right, dock.Top, dock.Bottom }.Where(n => n != null).ToArray();
```

Comment:
I do not know how to decide if dock should go to left/right/bottom/top. This is just for demonstration, so all 4 pins can be tested.

Reply:
We need to find solution for finding suitable dock before this gets merged.

Reply:
I thinks we can add hints to docks what is prefered pin dock direction.